### PR TITLE
feat: update the form post api

### PR DESCRIPTION
**Pull request by solittlework**
* POST body accepts only name and description
* id always derived from name via toKebabCase
* anonymousApply, applicantRoles, assessorRoles, clerkRoles, dataSchema hardcoded as server defaults
* FormDefinitionCreate schema added to swagger
* FormDefinition.id marked readOnly in swagger
* Stale default-role descriptions removed from swagger
* Obsolete tests removed, new edge-case tests added

### CODE REVIEW

File: apps/form-service/src/form/router/form.swagger.yml [hunk 1]
```diff
@@ -24,12 +24,12 @@ components:
           type: boolean
         applicantRoles:
           type: array
-          description: Roles allowed to submit the form. Defaults to ['urn:ads:platform:form-service:form-applicant'] when not provided.
+          description: Roles allowed to submit the form.
```
Comment (thdlam) [inline apps/form-service/src/form/router/form.swagger.yml:27]: remove as it is no longer in the request body

File: apps/form-service/src/form/router/form.swagger.yml [hunk 2]
```diff
@@ -24,12 +24,12 @@ components:
...
           items:
             type: string
         assessorRoles:
           type: array
-          description: Roles allowed to assess submitted forms. Defaults to [] when not provided.
+          description: Roles allowed to assess submitted forms.
```
Comment (thdlam) [inline apps/form-service/src/form/router/form.swagger.yml:32]: remove as it is no longer in the request body

**Review by thdlam — APPROVED** [review]
